__tests__/app.test.js

```diff
@@ -111,56 +111,82 @@ describe("GET /api/articles", () => {
+        .get("/api/articles?sort_by=article_title")
+        .expect(400)
+        .then((response) => {
+          expect(response.body.message).toBe("Bad request.");
+        });
+    });
+  })
+  describe("Order query", ()=>{
+    test("200: sends an array of articles, defaulting to sorting by the created_at date", () => {
+      return request(app)
       .get("/api/articles?order=asc")
       .expect(200)
       .then((response) => {
         const articlesArray = response.body.articles;
         expect(articlesArray).toBeSortedBy("created_at", {coerce: true});
       });
-  });
-  test("200: order query is case insensitive", () => {
-    return request(app)
+    });
+    test("200: order query is case insensitive", () => {
+      return request(app)
       .get("/api/articles?order=ASC")
       .expect(200)
       .then((response) => {
         const articlesArray = response.body.articles;
         expect(articlesArray).toBeSortedBy("created_at", {coerce: true});
       });
-  });
-  test("400: when an invalid sort-by query is sent through, responds with an error informing user", () => {
-    return request(app)
-      .get("/api/articles?sort_by=article_title")
-      .expect(400)
+    });
+    test("400: when an invalid order query is sent through, responds with an error informing user", () => {
+      return request(app)
+        .get("/api/articles?order=diagonal")
+        .expect(400)
+        .then((response) => {
+          expect(response.body.message).toBe("Bad request.");
+        });
+    });
+  })
+  describe("Topic query",() => {
+    test("200: filters the articles by the topic specified in query", () => {
+      return request(app)
+      .get("/api/articles?topic=mitch")
+      .expect(200)
       .then((response) => {
-        expect(response.body.message).toBe("Bad request.");
+        const articlesArray = response.body.articles;
+        expect(articlesArray.length).toBe(12)
+        articlesArray.forEach((article)=> {
+          expect(article.topic).toBe("mitch")
+        })
       });
-  });
-  test("400: when an invalid order query is sent through, responds with an error informing user", () => {
+    })
+    test("400: when an invalid topic query is sent through, responds with an error informing user", () => {
+      return request(app)
+        .get("/api/articles?topic=mitchell")
+        .expect(400)
+        .then((response) => {
+          expect(response.body.message).toBe("Bad request.");
+        });
+    });
+  })
+  test("200: sends an array of correctly formatted articles in the order of categories decided in query", () => {
     return request(app)
-      .get("/api/articles?order=diagonal")
-      .expect(400)
+      .get("/api/articles?sort_by=title&order=asc")
+      .expect(200)
       .then((response) => {
-        expect(response.body.message).toBe("Bad request.");
+        const articlesArray = response.body.articles;
+        expect(articlesArray).toBeSortedBy("title");
       });
   });
 });
```

Comment:
What about if I have a sort_by and order query? What about if I have all the queries at once? Let's test for these scenarios too.